packages/extension-core/src/domains/accounts/helpers.ts

```diff
@@ -137,8 +138,21 @@ export const isCurveCompatibleWithChain = (
 }
 
 export const isAccountCompatibleWithChain = (chain: Chain, account: Account) => {
+  if (account.type === "ledger-ethereum") return false
+
   const genesisHash = getAccountGenesisHash(account)
   if (genesisHash && genesisHash !== chain.genesisHash) return false
   if (isAccountLedgerPolkadotGeneric(account) && !chain.hasCheckMetadataHash) return false
-  return isAccountEthereum(account) ? chain.account === "secp256k1" : chain.account !== "secp256k1"
+  return isAccountAddressEthereum(account)
+    ? chain.account === "secp256k1"
+    : chain.account !== "secp256k1"
+}
+
+export const isAccountCompatibleWithNetwork = (
+  network: SimpleEvmNetwork | Chain,
+  account: Account,
+) => {
+  return /^\d+$/.test(network.id)
+    ? isAccountPlatformEthereum(account)
+    : isAccountCompatibleWithChain(network as Chain, account)
```

Comment:
I would strongly prefer if we didn't start testing whether chain/network ids are strings or numbers 😢
Also I don't think this function is currently used anywhere!

```suggestion
export const isAccountCompatibleWithNetwork = (network: SimpleEvmNetwork, account: Account) =>
  isAccountPlatformEthereum(account)
```

Reply:
good catch, forgot to delete it :)

Reply:
would be nice to have such a helper to be able to check if an account (which can be evm or substrate) can sign on a network (which can be evm or substrate), but it can wait until networks consolidation update.